src/build.ts

```diff
@@ -0,0 +1,186 @@
+
+    // For now we'll bundle all functions until this is available on a manifest
+    // TODO: add this check back once we add it to the manifest definition
+    // if (fnDef.runtime_environment !== 'slack') {
+    //   continue;
+    // }
+
+    //For API type functions, there are no function files.
+    if (fnDef.type === "API") {
+      continue;
+    }
+
+    // Always validate function paths
+    const fnFilePath = await getValidFunctionPath(
+      workingDirectory,
+      fnId,
+      fnDef,
+    );
+
+    await createFunctionFile(
+      outputDirectory,
+      fnId,
+      fnFilePath,
+      hookCLI,
+    );
+  }
+};
+
+const functionPathHasDefaultExport = async (
+  functionFilePath: string,
+) => {
+  const functionModule = await import(`file://${functionFilePath}`);
+  return functionModule.default
+    ? typeof functionModule.default == "function"
+    : false;
+};
+
+const getValidFunctionPath = async (
+  workingDirectory: string,
+  fnId: string,
+  // deno-lint-ignore no-explicit-any
+  fnDef: any,
+) => {
+  if (!fnDef.source_file) {
+    throw new Error(
+      `Run on Slack function provided for ${fnId}, but no source_file was provided.`,
+    );
+  }
+
+  const fnFilePath = path.join(workingDirectory, fnDef.source_file);
+
+  // Make sure it's a file that exists
+  try {
+    const { isFile } = await Deno.stat(fnFilePath);
+    if (!isFile) {
+      throw new Error(`Could not find file: ${fnFilePath}`);
+    }
+  } catch (e) {
+    if (e instanceof Deno.errors.NotFound) {
+      throw new Error(
+        `Could not find file: ${fnFilePath}. Make sure your function definition's "source_file" is relative to your project root.`,
+      );
+    }
+    throw new Error(e);
+  }
+
+  if (!await functionPathHasDefaultExport(fnFilePath)) {
+    throw new Error(
+      `File: ${fnFilePath}, containing your function does not define a default export handler.`,
+    );
+  }
+  return fnFilePath;
+};
+
+const createFunctionFile = async (
+  outputDirectory: string,
+  fnId: string,
+  fnFilePath: string,
+  hookCLI: Protocol,
+) => {
+  const fnFileRelative = path.join("functions", `${fnId}.js`);
+  const fnBundledPath = path.join(outputDirectory, fnFileRelative);
+
+  // We'll default to just using whatever Deno executable is on the path
+  // Ideally we should be able to rely on Deno.execPath() so we make sure to bundle with the same version of Deno
+  // that called this script. This is perhaps a bit overly cautious, so we can look to remove the defaulting here in the future.
+  let denoExecutablePath = "deno";
+  try {
+    denoExecutablePath = Deno.execPath();
+  } catch (e) {
+    hookCLI.error("Error calling Deno.execPath()", e);
+  }
+
+  try {
+    // call out to deno to handle bundling
+    const p = Deno.run({
+      cmd: [
+        denoExecutablePath,
+        "bundle",
+        fnFilePath,
+        fnBundledPath,
+      ],
+    });
+
+    const status = await p.status();
+    p.close();
```

Comment:
🧠

Reply:
When I was running tests a bunch of times, especially in quick succession, I would see an occasional error being thrown around still-open buffers to a dead process. After adding this and stress-testing it a bunch, that error seemed to go away.
Seems odd that the official deno docs don't show using `close()` in their example 😬